src/box/lua/config/configdata.lua

```diff
@@ -91,6 +91,9 @@ local function instance_sharding(iconfig, instance_name)
     end
     local zone = instance_config:get(iconfig, 'sharding.zone')
     local uri = instance_config:instance_uri(iconfig, 'sharding')
+    if uri == nil then
+        error('No suitable URI provided', 0)
+    end
```

Comment:
Nit: Shouldn't we report an instance name at least?

Reply:
Added instance name. Also, added a test.